src/core/src/sys/interactions.rs

```diff
@@ -185,6 +185,20 @@ impl Interactions {
         }
     }
 
+    /// Get all pair interactions
+    ///
+    /// Using this function, you lose information
+    /// about which interaction belongs to which pair of particles.
+    pub fn all_pairs(&self) -> Vec<&PairInteraction> {
+        let res: Vec<&PairInteraction> = self.pairs
+            .values() // value is a Vec<&PairInteraction>
```

Comment:
I am not sure here. I thought that the value of the tree is a `Vec<PairInteraction>` looking at the definition in the struct. `flat_map` produces an iterator over the elements, i.e. `PairInteraction`s. This is done for every entry in the tree and finally everything is collected into a single vector. Did I get this wrong?